### PR TITLE
#1 React company local header - implement and fix broken tests

### PR DESCRIPTION
**Pull request by debitan**
## Description of change

This PR implements the React company local header which was brought in with two previous PRs - https://github.com/uktrade/data-hub-frontend/pull/2647 & https://github.com/uktrade/data-hub-frontend/pull/2653

Because of the significant changes in architecture and the lack of functional testing around this part of the site I have split the work into two sister PRs. This one and https://github.com/uktrade/data-hub-frontend/pull/2659

This PR contains the architectural change and fixes the tests that it breaks. It also removes the tests that would become redundant with the more thorough tests that comes in #2659. There are also some minor changes to the two components, `FlashMessages` and `CompanyLocalHeader`, to fix edge case bugs that weren't discovered until the functional tests were in place. 

For context - I've undertaken this work to be able to add a new button to the company local header. So to avoid adding to the legacy nunjucks template I have ported this part of the site into React. 

## Test instructions

Navigate to any company page to see the header in action. 

## Screenshots

No visual changes. 

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to master?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/companies/apps/advisers/controllers/__test__/advisers.test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
-const buildMiddlewareParameters = require('../../../../../../../test/unit/helpers/middleware-parameters-builder.js')
+const buildMiddlewareParameters = require('../../../../../../../test/unit/helpers/middleware-parameters-builder')
 const config = require('../../../../../../config')
 const companyMock = require('../../../../../../../test/unit/data/companies/companies-house.json')
-const coreTeamMock = require('../../../../../../../test/unit/data/companies/one-list-group-core-team.json')
+const coreTeamMock = require('../../../../../../../test/sandbox/fixtures/v4/company/one-list-group-core-team.json')
```
Comment (debitan) [inline src/apps/companies/apps/advisers/controllers/__test__/advisers.test.js:4]: I had to move this fixture into the sandbox app so it could also be used there. Sandbox doesn't seem to be able to access files outside of its own folder structure.

File: src/client/components/LocalHeader/FlashMessages.jsx [hunk 2]
```diff
@@ -31,22 +32,31 @@ const messageColours = {
...
+          ))
+        : message.map((message) => (
+            <li key={message}>
+              <StatusMessage colour={messageColours[type]}>
+                <StyledMessage dangerouslySetInnerHTML={{ __html: message }} />
+              </StatusMessage>
+            </li>
+          ))
+    })}
+  </UnorderedList>
```
Comment (debitan) [inline src/client/components/LocalHeader/FlashMessages.jsx:58]: Here I am wrapping the flash messages in `li`s and also making sure to map over the multiple messages of the same type when they don't have a body.

File: src/client/components/LocalHeader/__stories__/FlashMessages.stories.jsx [hunk 3]
```diff
@@ -28,7 +28,7 @@ storiesOf('Flash Messages', module)
             body: 'Info message body',
           },
         ],
-        error: ['Error test message'],
+        error: ['Error test message', 'Another error message'],
```
Comment (debitan) [inline src/client/components/LocalHeader/__stories__/FlashMessages.stories.jsx:31]: This is here to show that the component can now show multiple messages of the same type when there is no body.

File: test/functional/cypress/specs/companies/activity-feed-spec.js [hunk 4]
```diff
@@ -1,80 +1,21 @@
...
-        expectedHeading
-      )
-    })
-
-    it('should display the address', () => {
-      cy.get(selectors.localHeader().headingAfter).should(
-        'have.text',
-        expectedAddress
-      )
-    })
```
Comment (debitan) [inline test/functional/cypress/specs/companies/activity-feed-spec.js:25]: These will be covered by new more thorough tests in the accompanying tests PR.

File: test/functional/cypress/specs/companies/company-global-ultimate-spec.js [hunk 5]
```diff
@@ -1,35 +0,0 @@
-const selectors = require('../../../../selectors')
```
Comment (debitan) [inline test/functional/cypress/specs/companies/company-global-ultimate-spec.js:1]: This is also covered in the new more thorough tests in the accompanying tests PR.

File: test/functional/cypress/specs/companies/investments/investment-projects-spec.js [hunk 6]
```diff
@@ -1,19 +0,0 @@
-const selectors = require('../../../../../selectors')
```
Comment (debitan) [inline test/functional/cypress/specs/companies/investments/investment-projects-spec.js:1]: Same here.

**Comment by debitan** [comment]
> A lot of work well done.
> I'll code wise looks good. I'll just check out the branch and run it locally to test it.
> My only comment would be to split the commits e.g advisers, activity-feed , investments could have be split into commit to make it easier to review.

Yeah sorry about that. The commit history got very messy and I had to unpick it a bunch to split it into the two PRs we've got. Am trying to get better with my commit discipline.

File: src/apps/companies/client/CompanyLocalHeader.jsx [hunk 7]
```diff
@@ -197,7 +197,7 @@ const CompanyLocalHeader = ({
       )}
 
       {company.pending_dnb_investigation && (
-        <StyledMain>
+        <StyledMain data-auto-id="investigationMessage">
```
Comment (mforner13) [inline src/apps/companies/client/CompanyLocalHeader.jsx:200]: Does this message currently look like this? It seems really odd just having the paragraph out on it's own.

File: src/templates/_layouts/template-no-local-header.njk [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {% extends "./template.njk" %}
 
-{% block local_header %}{% endblock %}
+{% block local_nav %}{% endblock %}
```
Comment (mforner13) [inline src/templates/_layouts/template-no-local-header.njk:3]: What was the reason behind this change? The Local Nav normally refers to these tabs in the nunjucks pages, so this could be misleading.  
Screenshot 2020-05-12 at 12 15 26

File: test/functional/cypress/specs/companies/lead-advisers/company-page-spec.js [hunk 9]
```diff
@@ -1,25 +1,14 @@
...
+const urls = require('../../../../../../src/lib/urls')
 
 describe('Lead advisers', () => {
   context('when viewing a non One List tier company', () => {
     before(() => {
-      cy.visit(companies.detail(fixtures.company.marsExportsLtd.id))
+      cy.visit(urls.companies.detail(fixtures.company.marsExportsLtd.id))
       cy.get(selectors.tabbedLocalNav().item(3)).click()
     })
-    it('should render breadcrumbs', () => {
```
Comment (mforner13) [inline test/functional/cypress/specs/companies/lead-advisers/company-page-spec.js:13]: Are the breadcrumb tests you've removed in this file covered elsewhere?

File: src/apps/companies/client/CompanyLocalHeader.jsx (hunk 7 above)
Comment (debitan) [reply src/apps/companies/client/CompanyLocalHeader.jsx:200]: good spot. I think it is supposed to look like this:

image

File: test/functional/cypress/specs/companies/lead-advisers/company-page-spec.js (hunk 9 above)
Comment (debitan) [reply test/functional/cypress/specs/companies/lead-advisers/company-page-spec.js:13]: Yes, they are covered in the second PR - #2659

File: src/templates/_layouts/template-no-local-header.njk (hunk 8 above)
Comment (debitan) [reply src/templates/_layouts/template-no-local-header.njk:3]: to be honest. I can't remember. I'll revert the change and see what happens...
